src/components/usePopup.ts

```diff
@@ -13,8 +16,9 @@ export type PopupProps = {
 }
 
 export const usePopup = (): PopupProps => {
-    const [transferSize, setTransferSize] = useState(0);
-    const [selectedCountries, setSelectedCountries] = useState<Map<CountryName, number>>(new Map<CountryName, number>())
+    const [totalBytesReceived, setTotalBytesReceived] = useState(0);
+    const [emissions, setEmissions] = useState(0);
+    const [selectedCountries, setSelectedCountries] = useState<Map<CountryName, number>>(new Map<CountryName, number>([["World Average", 0]]))
```

Comment:
I think we can substitute out the `Map` object with an array of objects e.g. `useState<Record<string, number>[]>()` - this is because we conventionally avoid forEach functions and let keywords in React apps and substitute them for functions e.g. reduce and map

Reply:
Could I clarify the following points:
- Is the avoidance of `let` to enforce immutability of state variables? Or is there some other reason
- Why are forEach functions avoided?

Reply:
Yeah we try and minimise `let` keywords and enforce consts as much as we can because we use arrays in as many places as possible, which means that we can use just the `.map()` and other array functions to modify an existing array and return a new one

Reply:
> Yeah we try and minimise `let` keywords and enforce consts as much as we can because we use arrays in as many places as possible, which means that we can use just the `.map()` and other array functions to modify an existing array and return a new one



I tried to use 
`const [selectedCountries, setSelectedCountries] = useState<Record<CountryName, number>>({
        "World Average": 0,
    })` 

but it showed a linting error: `Argument of type '{ "World Average": number; }' is not assignable to parameter of type 'Record<"World Average" | "United Kingdom" | "Austria" | "Belgium" | "Bulgaria" | "Croatia" | "Cyprus" | "Czech Republic" | "Denmark" | "Estonia" | "Finland" | "France" | "Germany" | ... 37 more ... | "Brazil", number> | (() => Record<...>)'.
  Type '{ "World Average": number; }' is missing the following properties from type 'Record<"World Average" | "United Kingdom" | "Austria" | "Belgium" | "Bulgaria" | "Croatia" | "Cyprus" | "Czech Republic" | "Denmark" | "Estonia" | "Finland" | "France" | "Germany" | ... 37 more ... | "Brazil", number>': "United Kingdom", Austria, Belgium, Bulgaria, and 46 more.`

On the other hand `const [selectedCountries, setSelectedCountries] = useState<Record<string, number>>({
        "World Average": 0,
    })`  works fine

Does this mean we have to drop the typing of the key to ensure that it exists in the `CO2_EMISSIONS_GRAMS_PER_GB` dictionary? 

Currently it is initialised with `export type CountryName = keyof typeof CO2_EMISSIONS_GRAMS_PER_GB;`, and Henry and I decided to do this so that the keys used to store selected countries will always exist in the CO2_EMISSIONS_GRAMS_PER_GB dictionary